### PR TITLE
CORTX-29828: Unit Tests: Drive state change broadcast and KV update

### PR DESCRIPTION
**Pull request by supriyachavan4398**
This is a unit test to check the behavior of drive state failure.
We invoke the handler for a device_state_set. We also check that the
failure is broadcast to motr.

Signed-off-by: Supriya Yadav <supriya.s.chavan@seagate.com>

### CODE REVIEW

**Comment by supriyachavan4398** [comment]
Pytest is working fine, If I run the test myself, it gives the following output:
```
(.py3venv) [root@ssc-vm-g2-rhev4-1947 unittest]# pytest -v -s -rP hax/test/test_failure.py
======================================================================= test session starts ========================================================================
platform linux -- Python 3.6.8, pytest-7.0.1, pluggy-1.0.0 -- /root/cortx_29546/unittest/.py3venv/bin/python3.6
cachedir: .pytest_cache
rootdir: /root/cortx_29546/unittest/hax
plugins: aiohttp-0.3.0, cov-3.0.0, mock-3.6.1, timeout-1.4.2
collected 2 items

hax/test/test_failure.py::TestFailure::test_drive_failure PASSED
hax/test/test_failure.py::TestFailure::test_process_failure PASSED

============================================================================== PASSES ==============================================================================
__________________________________________________________________ TestFailure.test_drive_failure __________________________________________________________________
------------------------------------------------------------------------ Captured log call -------------------------------------------------------------------------
DEBUG    hax:__init__.py:76 HA broadcast, node: ssc-vm-g2-rhev4-1947.colo.seagate.com device: /dev/sdd state: failed
DEBUG    hax:__init__.py:297 Broadcasting HA states [HAState(fid=0x6b00000000000001:0x11, status=FAILED)] over ha_link
_________________________________________________________________ TestFailure.test_process_failure _________________________________________________________________
------------------------------------------------------------------------ Captured log call -------------------------------------------------------------------------
DEBUG    hax:__init__.py:297 Broadcasting HA states [HAState(fid=0x7200000000000001:0x15, status=FAILED)] over ha_link
DEBUG    hax:__init__.py:520 Process fid=0x7200000000000001:0x15 encloses 1 services as follows: [FidWithType(fid=0x7300000000000001:0xe, service_type='ios')]
TRACE    hax:cache.py:79 CACHE: created. fn_name=Motr._generate_sub_disks
DEBUG    hax:__init__.py:557 proc fid=0x7200000000000001:0x15 encloses 1 disks as follows: [0x6b00000000000001:0x11]
DEBUG    hax:__init__.py:641 Notifying node status for process_fid=0x7200000000000001:0x15 state=ObjHealth.FAILED
DEBUG    hax:__init__.py:604 node_fid: 0x6e00000000000001:0x3 encl_fid: 0x6500000000000001:0x4 ctrl_fids: [0x6300000000000001:0x5] with state: ObjHealth.FAILED
======================================================================== 2 passed in 0.13s =========================================================================
```
Testing is in progress for LC env.

File: hax/test/test_failure.py [hunk 1]
```diff
@@ -144,3 +148,52 @@ def test_process_failure(self):
...
+        # Send the mock event for drive failure.
+        bqprocessor.handle_device_state_set(payload)
+
+        # ConsulUtil is responsible for the actual KV updates, just check
+        # here that the appropriate util function is called for drive state
+        # update.
+        consul_util.update_drive_state.assert_called_with(
+                [drive_fid],
+                ObjHealth.FAILED,
+                kv_cache=consul_cache)
```
Comment (Shreya-18) [inline hax/test/test_failure.py:199]: @supriyachavan4398 please could you add asserts for what is being broadcasted. 

Similar to above `test_process_failure` UT

```
broadcast_list = motr._ha_broadcast.call_args[0][0]
self.assertTrue(_has_failed_note(broadcast_list, drive_fid))
```

File: hax/test/test_failure.py [hunk 2]
```diff
@@ -144,3 +148,52 @@ def test_process_failure(self):
...
+        
+        svc_name = 'hax'
+        drive_fid = Fid(0x6b00000000000001, 0x11)
+        hax_fid = Fid(0x7200000000000001, 0x6)
+        
+        def fake_add(cmd):
+            if isinstance(cmd, BroadcastHAStates):
+                motr.broadcast_ha_states(cmd.states, kv_cache=consul_cache)
+            if hasattr(cmd, 'reply_to') and cmd.reply_to:
+                cmd.reply_to.put([MessageId(1, 42)])
```
Comment (Shreya-18) [inline hax/test/test_failure.py:171]: What is MessageId(1,42) ?

**Review by Shreya-18 — CHANGES_REQUESTED** [review]

File: hax/test/test_failure.py (hunk 1 above)
Comment (supriyachavan4398) [reply hax/test/test_failure.py:199]: Added. Thanks for reminding me!

**Comment by pavankrishnat** [comment]
run vm-deployment

**Comment by pavankrishnat** [comment]
run vm-deployment

**Comment by pavankrishnat** [comment]
rerun vm-deployment

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

**Comment by pavankrishnat** [comment]
Jenkins please retry a build

**Review by Shreya-18 — APPROVED** [review]

**Review by mssawant — APPROVED** [review]

**Pull request merged**